### ISSUE TITLE
open-vm-tools does not resize the display on Ubuntu 15.04

### ISSUE THREAD

**Issue by Melab** [vmware/open-vm-tools#54]
I am running Ubuntu 15.04 with Linux kernel release 4.2.0-29-generic on VMware Workstation 12 Player 12.1.0 build-3272444. I have installed the packages `open-vm-tools` and `open-vm-tools-desktop`. I have rebooted the virtual machine. I logged in and waited for the GUI to load completely. I maximized the window and the guest did not resize itself. I thought open-vm-tools can do autofit.

**Comment by ravindravmw** [comment]
Could you please check if you have "vmtoolsd -n vmusr" process running in the guest? And, if it has loaded libresolutionSet.so (lsof -p <pid> | grep libresolution)?

**Comment by Melab** [comment]
I have uninstalled both packages, so no, it is not running.

**Comment by Melab** [comment]
Any updates?

**Comment by ravindravmw** [comment]
> Your command gives lsof: no process ID specified.

Ah, looks like github ate the <PID> argument I mentioned. You need to specify "lsof -p _PID_ | grep libresolution" where _PID_ is the PID for vmtoolsd running as vmusr.

Given that you don't have vmtoolsd running as vmusr, above command is irrelevant. And, this is the reason why you are missing screen resizing.

Assuming you are using the packages provided by Ubuntu, you will need to raise this issue of "'vmtoolsd -n vmusr' not running in the VM" with Ubuntu.

**Comment by chiusite** [comment]
So, how do i fix?  typing sudo vmtoolsd --name=vmuser   did not create the process -- i thought it was what vmware-user-suid-wrapper is supposed to do (but it runs into the error that cannot open /proc/fs/vmblock/fs

**Comment by chiusite** [comment]
ok, i missed the installation of open-vm-tools-desktop.  No where in this website talks about installation much (kind of difficult when there are so many linux guest flavors)  Only vmware's KB documented it and I missed it because I wanted to get into open-vmtools instead of theirs.  

Now my linux guest screen can resize.  The unity mode still can be not entered (vmware complained / laughed at no unity support at guest msg).  

I hope that I dont have to choose between VirualBox's seamless/unity mode and  Vmware's phyical partition support.  I want both.

**Comment by ravindravmw** [comment]
That's good to know. Unfortunately, Unity support never made into open-vm-tools. Also, this feature is on deprecation path for VMware Tools on Linux. So, for this feature, the only option seems to be using older version of VMware Tools provided by the host.

**Comment by changyukang** [comment]
where is the open-vm-tools? I can not find it in archlinux AUR or official repository. Besides, I searched it on the internet without any useful guide.

**Comment by jmaurice** [comment]
+1 for needing "--name=vmusr" argument in vmtoolsd

**Comment by juj** [comment]
Aand right after typing, I find the `View -> Autosize -> Autofit Guest` menu option that was not enabled by default, but it was set to state `View -> Autosize -> Center Guest`. (I wonder if `Autofit Guest` would be a more sensible default?)

Enabling Autofit Guest resolves this for me, so I don't have an issue after all.

**Comment by iav** [comment]
Have same error with LinuxMint 18 in VM in esxi 5.5
/usr/lib/open-vm-tools/plugins/vmusr/libresolutionSet.so loaded, tries to use view menu of vmware client just shrink window to 800x600.
Is it possible to have other than 800x600 display resolution?

**Comment by ravindravmw** [comment]
@iav Probably, your VM does not have enough video memory to provide that larger resolution. Could you please search for "Max Resolution" in your VM's vmware.log file (can be found in the VM's directory) and let us know what you find?

**Comment by iav** [comment]
'''
2016-07-15T12:45:42.918Z| vmx| I120: SVGA: Device capabilities 0x00ff83e2
2016-07-15T12:45:42.918Z| vmx| I120: SVGA: Number of displays 1
2016-07-15T12:45:42.918Z| vmx| I120: SVGA: Maximum display topology 2560x1600.
2016-07-15T12:45:42.918Z| vmx| I120: SVGA: Truncated maximum resolution to VRAM size: 4194304 bytes VRAM, 1176x885 Max Resolution
2016-07-15T12:45:42.918Z| vmx| I120: SVGA: FIFO capabilities 0x0000077f
'''

**Comment by iav** [comment]
Now I set video memory to 9 MB. Nothing changes.

**Comment by iav** [comment]
No, now, with added memory, "fit guest now" start to work!
Thank you!

**Comment by ravindravmw** [comment]
> 2016-07-15T12:45:42.918Z| vmx| I120: SVGA: Truncated maximum resolution to VRAM size: 4194304 bytes VRAM, 1176x885 Max Resolution

TIP: If there is an attempt to increase the resolution above max possible, it will default to 800x600. The trick to find max possible resolution (without looking at log file) is to increase the resolution in smaller steps and the moment if falls back to 800x600 you know you have hit the limit. You can go back to previous working resolution the moment that happens that way you would be able to get to max possible resolution. This tip works when you don't have any configuration issues.

**Comment by JoeyG1973** [comment]
I am running Ubuntu 16.04 Gnome on VMware Workstation 12 Pro 12.1.1 build-3770994 with open vm tools 10.0.7.52125 (build-3227872).  Screen resizing works just fine even in full screen and when I cycle multiple monitors.  I have 2 monitors.  When I turn off multiple monitors the screen resizing stops working.  The screen stays the same size as though it has multiple monitors and the mouse is WAY off in location.

Here is what is in the vmware.log file when I cycle from 2 screens to 1 screen them back to 2 screens:

`2016-08-05T13:26:12.153-04:00| mks| I125: ToolsSetDisplayTopology: Sending rpcMsg = DisplayTopology_Set 1 , 0 0 1920 1200
2016-08-05T13:26:12.290-04:00| mks| I125: MKS-SWB: Number of MKSWindows changed: 1 rendering MKSWindow(s) of total 3.
2016-08-05T13:26:12.363-04:00| mks| I125: MKS-SWB: Number of MKSWindows changed: 2 rendering MKSWindow(s) of total 3.
2016-08-05T13:26:18.986-04:00| mks| I125: ToolsSetDisplayTopology: Sending rpcMsg = DisplayTopology_Set 2 , 0 0 1920 1200 , 1920 0 1920 1200
2016-08-05T13:26:19.259-04:00| svga| I125: MKS-SWB: Number of MKSWindows changed: 1 rendering MKSWindow(s) of total 2.
2016-08-05T13:26:19.332-04:00| mks| I125: MKS-SWB: Number of MKSWindows changed: 2 rendering MKSWindow(s) of total 3.`

Any ideas here?

Joe

**Comment by asangas** [comment]
@JoeyG1973 did you have any luck with your problem? I'm having the exact same issue.

**Comment by JoeyG1973** [comment]
Nope still does it.

**Comment by asangas** [comment]
Finally I found a work-around. Downgrading to kernel 4.3.6 works for me beautifully on 1604, as discussed in this issue: https://github.com/vmware/open-vm-tools/issues/74

**Comment by devarni** [comment]
Same problem.. Debian stable (Jessie) as guest on Windows 10 64Bit... since 2 days I cannot resize the display to fullscreen.
The last entry in the X log shows: "Requested framebuffer size 1302x881x32 will not fit in display memory"

3D acceleration is disabled (I don't need it), I've tried it also with enabled acceleration and set the recommended memory without luck.

**Comment by EMCP** [comment]
For future reference, you have to switch the VM's into Auto Settings for their GPUs.. and then I got fulscreen working post install of `open-vm-tools-desktop`

In the ESXi Server : 

![image](https://user-images.githubusercontent.com/3691722/46611733-e5c9fa00-cb0e-11e8-8a13-c09d084d1dc5.png)

**Comment by justdave** [comment]
I was running into this on Ubuntu 18.10, and since Google searches point here and nobody's mentioned it here yet...   I had both `open-vm-tools` and `open-vm-tools-desktop` already installed and it was still behaving like this (failing to detect the resolution).  I fixed it by running `sudo apt-get install --reinstall open-vm-tools`

**Comment by farmerbb** [comment]
@justdave This fixes the issue temporarily for me, but after a reboot of the guest VM I have to run that command to fix it again.

**Comment by justdave** [comment]
Yeah, I found the same.  I found an askubuntu at https://askubuntu.com/questions/1085621/vmware-workstation-ubuntu-18-10-full-screen-problem that suggested adding a line to the systemd config file for the service to change its launch order...

edit `/etc/systemd/system/multi-user.target.wants/open-vm-tools.service` and add this line at the end of the "Unit" section:
```
After=graphical.target
```
This indeed makes it come back after a reboot, but there's an annoying flash because the desktop comes up at 860x480 or somesuch and then changes to the full resolution 5 or 10 seconds later. It also fails to retain the scaling setting.  I have a retina display so I have to go change the scaling to 200% after rebooting every time to be able to read anything.
